File: Lucidity/FlashesForm.vb
Public Class FlashesForm

  Private m_intPosition As Integer
  Public FlashColor As Color
  Public BGColor As Color
  Public Timeout As Integer = 1000
  Private m_boolOn As Boolean = False


  Public Sub New(ByVal Interval As Integer)
    InitializeComponent()

    Me.Width = Screen.PrimaryScreen.Bounds.Width
    Me.Height = Screen.PrimaryScreen.Bounds.Height

    tmrLoad.Interval = Interval
  End Sub

  Public Sub Play()
    tmrLoad.Enabled = True
  End Sub

  Private Sub tmrLoad_Tick(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles tmrLoad.Tick
    tmrLoad.Enabled = False
    If m_boolOn Then
      m_boolOn = False
      lblText.BackColor = BGColor
    Else
      m_boolOn = True
      lblText.BackColor = FlashColor
    End If

    m_intPosition += 1

    If m_intPosition >= Timeout Then
      tmrLoad.Enabled = False
      Close()
      Return
    End If
    tmrLoad.Enabled = True
  End Sub

  Private Sub lblText_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles lblText.Click
    Close()
  End Sub

  Private Sub FlashesForm_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Click
    Close()
  End Sub

  Private Sub SpeedReadingForm_FormClosing(ByVal sender As Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles Me.FormClosing
    Cursor.Show()
  End Sub

  Private Sub SpeedReadingForm_KeyDown(ByVal sender As System.Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles MyBase.KeyDown
    If e.KeyCode = Keys.Escape Then
      Close()
    End If

    tmrLoad.Enabled = Not tmrLoad.Enabled
  End Sub

  Private Sub SpeedReadingForm_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Load
    Cursor.Hide()
  End Sub
End Class

File: Lucidity/FlashesForm.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
namespace lucidcode.LightenedDream.Lucidity
{
	public partial class FlashesForm
	{

		private int m_intPosition;
		public Color FlashColor;
		public Color BGColor;
		public int Timeout = 1000;

		private bool m_boolOn = false;

		public FlashesForm(int Interval)
		{
			Load += SpeedReadingForm_Load;
			KeyDown += SpeedReadingForm_KeyDown;
			FormClosing += SpeedReadingForm_FormClosing;
			Click += FlashesForm_Click;
			InitializeComponent();

			this.Width = Screen.PrimaryScreen.Bounds.Width;
			this.Height = Screen.PrimaryScreen.Bounds.Height;

			tmrLoad.Interval = Interval;
		}

		public void Play()
		{
			tmrLoad.Enabled = true;
		}

		private void tmrLoad_Tick(System.Object sender, System.EventArgs e)
		{
			tmrLoad.Enabled = false;
			if (m_boolOn) {
				m_boolOn = false;
				lblText.BackColor = BGColor;
			} else {
				m_boolOn = true;
				lblText.BackColor = FlashColor;
			}

			m_intPosition += 1;

			if (m_intPosition >= Timeout) {
				tmrLoad.Enabled = false;
				Close();
				return;
			}
			tmrLoad.Enabled = true;
		}

		private void lblText_Click(System.Object sender, System.EventArgs e)
		{
			Close();
		}

		private void FlashesForm_Click(object sender, System.EventArgs e)
		{
			Close();
		}

		private void SpeedReadingForm_FormClosing(object sender, System.Windows.Forms.FormClosingEventArgs e)
		{
			Cursor.Show();
		}

		private void SpeedReadingForm_KeyDown(System.Object sender, System.Windows.Forms.KeyEventArgs e)
		{
			if (e.KeyCode == Keys.Escape) {
				Close();
			}

			tmrLoad.Enabled = !tmrLoad.Enabled;
		}

		private void SpeedReadingForm_Load(object sender, System.EventArgs e)
		{
			Cursor.Hide();
		}
	}
}
